src/nvim/api/extmark.c

```diff
@@ -552,9 +552,9 @@ Integer nvim_buf_set_extmark(Buffer buffer, Integer ns_id, Integer line, Integer
   colnr_T col2 = -1;
   if (HAS_KEY(opts, set_extmark, end_col)) {
     Integer val = opts->end_col;
-    VALIDATE_RANGE((val >= 0 && val <= MAXCOL), "end_col", {
```

Comment:
why not validate it still?

    VALIDATE_RANGE((val >= -1 && val <= MAXCOL), "end_col", {

Reply:
Makes sense, it's done.